proof/encoding.go

```diff
@@ -110,6 +128,10 @@ func SplitRootProofEncoder(w io.Writer, val any, buf *[8]byte) error {
 }
 
 func SplitRootProofDecoder(r io.Reader, val any, buf *[8]byte, l uint64) error {
+	if l > tlv.MaxRecordSize+asset.MaxAssetEncodeSize {
```

Comment:
Would you mind please adding a comment here to explain why we use `MaxAssetEncodeSize` at this point.

Reply:
IIUC it's because the split root is just the full root asset right?

Reply:
Hmm, actually, taking a second look, this is just a `TaprootProof`, so it should have the same max size as that record. The thinking was around the split root asset, yes. But that's in the split commitment within previous witness, not in the split root proof.